examples/pytorch/mnist_trainer.py

```diff
@@ -0,0 +1,112 @@
+    _data_module = MNISTDataModule(batch_size=32)
+    trainer = Trainer(
+        accelerator=HPUAccelerator(),
+        devices=1,
+        strategy=SingleHPUStrategy(),
+        plugins=plugin,
+        fast_dev_run=True,
+    )
+    trainer.fit(model, _data_module)
+    trainer.test(model, _data_module)
+
+
+def check_and_init_plugins(plugins, run_type, verbose):
+    """Initialise plugins with appropriate checks."""
+    _plugins = []
+    for plugin in plugins:
+        if verbose:
+            print(f"Initializing {plugin}")
+        if plugin == "MixedPrecisionPlugin":
+            warnings.warn("Operator overriding is not supported with MixedPrecisionPlugin on Habana devices.")
+            if run_type != "autocast":
+                _plugins.append(MixedPrecisionPlugin(device="hpu", precision="bf16-mixed"))
+            else:
+                warnings.warn("Skipping MixedPrecisionPlugin. Redundant with autocast run.")
+        else:
+            print(f"Unsupported or invalid plugin: {plugin}")
+    return _plugins
+
+
+def run_model(run_type, plugins, verbose):
+    """Picks appropriate model and plugins."""
+    # Initialise plugins
+    _plugins = check_and_init_plugins(plugins, run_type, verbose)
+    if run_type == "basic":
+        _model = LitClassifier()
+    elif run_type == "autocast":
+        if "LOWER_LIST" in os.environ or "FP32_LIST" in os.environ:
+            _model = LitAutocastClassifier(op_override=True)
+        else:
+            _model = LitAutocastClassifier()
+        warnings.warn(
+            "To override operators with autocast, set LOWER_LIST and FP32_LIST file paths as env variables."
+            "Example: LOWER_LIST=<path_to_bf16_ops> python example.py"
+            "https://docs.habana.ai/en/latest/PyTorch/PyTorch_Mixed_Precision/Autocast.html#override-options"
+        )
+
+    if verbose:
+        print(f"With run type: {run_type}, running model: {_model} with plugin: {_plugins}")
+    return run_trainer(_model, _plugins)
+
+
+def parse_args():
+    """Cmdline arguments parser."""
+    parser = argparse.ArgumentParser(description="Example to showcase mixed precision training with HPU.")
+
+    parser.add_argument("-v", "--verbose", action="store_true", help="Enable verbosity")
+    parser.add_argument(
+        "-r", "--run_types", nargs="+", choices=RUN_TYPE, default=RUN_TYPE, help="Select run type for example"
+    )
+    parser.add_argument(
+        "-p", "--plugins", nargs="+", default=[], choices=["MixedPrecisionPlugin"], help="Plugins for use in training"
+    )
+    return parser.parse_args()
+
+
+if __name__ == "__main__":
+    # Get options
+    options = parse_args()
+    if options.verbose:
+        print(f"Running MNIST mixed precision training with options: {options}")
+
+    # Run model and print accuracy
+    for run_type in options.run_types:
+        seed_everything(42)
+        run_model(run_type, options.plugins, options.verbose)
```

Comment:
why not use Lightning CLI?

Reply:
tested with LightningCLI, currently it does not seem to support setting accelerator class in trainer config. Setting it with "hpu" picks HPUAccelerator from lightning and not from lightning_habana

Reply:
Cc: @awaelchli

Reply:
@Borda @assassin1991 lets create an issue and track it through for CLI and external accelerators

Reply:
Issue created: https://github.com/Lightning-AI/lightning-Habana/issues/55